source/common/quic/envoy_quic_dispatcher.cc

```diff
@@ -107,5 +108,26 @@ quic::QuicConnectionId EnvoyQuicDispatcher::ReplaceLongServerConnectionId(
   return new_connection_id;
 }
 
+void EnvoyQuicDispatcher::closeConnectionsWithFilterChain(
+    const Network::FilterChain* filter_chain) {
+  if (connections_by_filter_chain_.find(filter_chain) != connections_by_filter_chain_.end()) {
+    std::list<std::reference_wrapper<Network::Connection>>& connections =
+        connections_by_filter_chain_[filter_chain];
+    // Retain the number of connections in the list early because closing the connection will change
+    // the size.
+    size_t num_connections = connections.size();
+    for (size_t i = 0; i < num_connections; ++i) {
```

Comment:
Should this be `while (!connections.empty())`?

Reply:
Yup, that's fewer lines of code.

Reply:
Reverted. It turns out to cause use-after-free. I forgot to mention that `connections` will be destroyed at the end of the loop. So the loop has to be written in this way.